vscode/src/webviewPanel.ts

```diff
@@ -390,38 +395,65 @@ export function registerWebViewCommands(context: ExtensionContext) {
   );
 }
 
+type PanelDesc = {
+  title: string;
+  panel: QSharpWebViewPanel;
+  state: any;
+};
+
 type PanelType =
   | "histogram"
   | "estimates"
   | "help"
   | "circuit"
   | "documentation";
 
-const panelTypeToPanel: Record<
-  PanelType,
-  { title: string; panel: QSharpWebViewPanel | undefined; state: any }
-> = {
-  histogram: { title: "Q# Histogram", panel: undefined, state: {} },
-  estimates: { title: "Q# Estimates", panel: undefined, state: {} },
-  circuit: { title: "Q# Circuit", panel: undefined, state: {} },
-  help: { title: "Q# Help", panel: undefined, state: {} },
-  documentation: {
-    title: "Q# Documentation",
-    panel: undefined,
-    state: {},
-  },
+const panels: Record<PanelType, { [id: string]: PanelDesc }> = {
+  histogram: {},
+  estimates: {},
+  circuit: {},
+  help: {},
+  documentation: {},
 };
 
-export function sendMessageToPanel(
-  panelType: PanelType,
-  reveal: boolean,
-  message: any,
-) {
-  const panelRecord = panelTypeToPanel[panelType];
-  if (!panelRecord.panel) {
-    const panel = window.createWebviewPanel(
+const panelTypeToTitle: Record<PanelType, string> = {
+  histogram: "Histogram",
+  estimates: "Q# Estimates",
+  circuit: "Circuit",
+  help: "Q# Help",
+  documentation: "Q# Documentation",
+};
+
+function getPanel(type: PanelType, id: string): PanelDesc | undefined {
+  return panels[type][id];
+}
+
+export function isPanelOpen(panelType: PanelType, id: string): boolean {
+  return getPanel(panelType, id)?.panel !== undefined;
+}
+
+function createPanel(
+  type: PanelType,
+  id: string,
+  webViewPanel?: WebviewPanel,
+): PanelDesc {
+  if (webViewPanel) {
+    const title = webViewPanel.title;
+    const panel = new QSharpWebViewPanel(type, webViewPanel, id);
+    panels[type][id] = { title, panel, state: {} };
+    return panels[type][id];
+  } else {
+    const title =
+      type == "help"
+        ? "Q# Help"
+        : type == "documentation"
+          ? "Q# Documentation"
+          : type == "estimates"
+            ? "Q# Estimates"
+            : `${id} ${panelTypeToTitle[type]}`;
```

Comment:
I think a title like `Q# Histogram - NameOfFile.qs` would look more consistent with the other tabs. 

![Screenshot 2025-01-24 134235](https://github.com/user-attachments/assets/5cec005a-b5f5-4ba0-abc9-2fc6c2c606f7)

Reply:
done